### PR TITLE
Start session

### PR DESCRIPTION
**Pull request by davidpn11**
Create a session on firebase. Involves:
- Add redux store
- Add session part on redux store
- API methods to create session and non-working others
- Firestore normalization methods
- Atempt to persist redux session
- Room renaming
-Lobby room

TODO: 
- Create session code method
- Make lobby work by /:id

### CODE REVIEW

**Comment by davidpn11** [comment]
TODO:
- Shuffle deck

File: src/api/db/session.ts [hunk 1]
```diff
@@ -1,13 +1,114 @@
...
+export async function requestCreateSession(
+  adminName: string
+): Promise<SessionWithId> {
+  const s = {
+    code: codeGenerator(),
+    status: "INITIAL" as NoGameSession["status"],
+    admin: adminName,
+  };
+
+  console.log("go");
```
Comment (mikelfcosta) [inline src/api/db/session.ts]: Console.log

File: src/model/Session.ts [hunk 2]
```diff
@@ -3,43 +3,62 @@ import { Card } from "./Card";
...
 
-type Play = {
+export type Play = {
   player: Player;
   type: "PLAY_CARD" | "DRAW_CARD" | "ACTION";
   card: Option<Card>;
   target: Option<Player>;
 };
+
+export type DocumentSnapshot = firebase.firestore.DocumentSnapshot<
```
Comment (mikelfcosta) [inline src/model/Session.ts]: Move this to the firebase models

File: src/api/db/session.ts [hunk 3]
```diff
@@ -1,13 +1,114 @@
...
-  } catch (error) {
-    console.error(error);
+    return data;
+  } else {
+    return {};
   }
 }
+
+//TODO: IMPROVE THIS CODE
+export async function requestCreateSession(
```
Comment (mikelfcosta) [inline src/api/db/session.ts:43]: Put a Try/Catch on it's contents, none of the functions has one, we could have at least one in this top one.

File: src/components/Button/index.tsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { ButtonVariant, ButtonWrapper } from "./style";
 type Props = {
   children: React.ReactNode;
   variant: ButtonVariant;
+  onClick?: () => void;
```
Comment (mikelfcosta) [inline src/components/Button/index.tsx:7]: Thanks

File: src/store/actions.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    payload: player,
+  };
+}
+
+export function createGameSession(name: string) {
+  return async (dispatch: Dispatch) => {
+    try {
+      const session = await requestCreateSession(name);
+      dispatch(setGameSession(session));
+      // const player = await requestAddPlayer(session.id, {
```
Comment (mikelfcosta) [inline src/store/actions.ts]: This comment is all necessary to stay?

**Review by mikelfcosta — APPROVED** [review]

File: src/store/actions.ts (hunk 5 above)
Comment (davidpn11) [reply src/store/actions.ts]: I removed part or it. This method might change in the future.

File: src/api/db/session.ts (hunk 3 above)
Comment (davidpn11) [reply src/api/db/session.ts:43]: This is done in the redux actions =)

**Pull request merged**